docs/source/reference/index.rst

```diff
@@ -1,6 +1,55 @@
 API Reference
 =============
 
+
+How to read Optuna API documentation?
+-------------------------------------
+
+We provide the API reference that is automatically generated by
+docstrings written in each modules. We basically follow `Google Style Python Docstrings <https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html>`_.
+For more information, please read our `Coding Style Conventions <https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#docstrings>`_ as well.
+A typical example of our docstring looks like following:
+
+.. code-block:: python
+
+    def hoge(self, number: int, ...) -> int:
+        """Description of a method
+
+        More detailed explanation for the API...
+        ...
+
+        Example:
+
+            .. testcode::
+
+                your_awesome_number = 1
+                hoge(your_awesome_number)
+
+                ...
+
+        Args:
+            number:
+                Some description for the argument here
+            ...
+
+        Raises:
+            TypeError:
+                Describe here when an exception is raised.
```

Comment:
This explanatory text seems to be targeting developers. How about targeting users as it's part of the library documentation? E.g. "Description of common error."